x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/reducer.ts

```diff
@@ -85,27 +93,24 @@ export const allRulesReducer = (
       };
     }
     case 'updateRules': {
-      if (state.rules != null) {
```

Comment:
Removed this as at no point is it typed or set to `null` or `undefined`, it's defaulted to empty array.